src/openfermion/hamiltonians/_general_hubbard_test.py

```diff
@@ -0,0 +1,172 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Tests for multiband Hubbard model module."""
+
+from collections import defaultdict
+import itertools
+import pytest
+import random
+
+
+from openfermion.hamiltonians import fermi_hubbard, FermiHubbardModel
+from openfermion.utils import HubbardSquareLattice, SpinPairs
+from openfermion.hamiltonians._general_hubbard import InteractionParameter
+
+
+def fermi_hubbard_from_general(x_dimension, y_dimension, tunneling, coulomb,
+                  chemical_potential=0.,
+                  periodic=True, spinless=False):
+    lattice = HubbardSquareLattice(x_dimension, y_dimension, 
+                             periodic=periodic, spinless=spinless)
+    interaction_edge_type = 'neighbor' if spinless else 'onsite'
+    model = FermiHubbardModel(lattice, 
+            tunneling_parameters=(('neighbor', (0, 0), tunneling),),
+            interaction_parameters=((interaction_edge_type, (0, 0), coulomb),),
+            potential_parameters=((0, chemical_potential),))
+    return model.hamiltonian()
+
+
+@pytest.mark.parametrize(
+        'x_dimension,y_dimension,tunneling,coulomb,' + 
+        'chemical_potential,spinless,periodic',
+        itertools.product(
+            range(1, 4), range(1, 4),
+            (random.uniform(0, 2.),), (random.uniform(0, 2.),),
+            (random.uniform(0, 2.),), (True, False), (True, False))
+        )
+def test_fermi_hubbard_square_special_general_equivalance(
+        x_dimension, y_dimension, tunneling, coulomb,
+        chemical_potential, spinless, periodic):
+    hubbard_model_special = fermi_hubbard(
+            y_dimension, x_dimension, tunneling, coulomb,
+            chemical_potential=chemical_potential, spinless=spinless,
+            periodic=periodic)
```

Comment:
The function `fermi_hubbard` also has arguments `magnetic_field` and `particle_hole_symmetry`. Do you intend to incorporate those into the general model?